circuits/cpp/src/aztec3/constants.hpp

```diff
@@ -184,8 +188,89 @@ enum GeneratorIndex {
     PRIVATE_CIRCUIT_PUBLIC_INPUTS,  // Size = 45
     PUBLIC_CIRCUIT_PUBLIC_INPUTS,   // Size = 32 (unused)
     FUNCTION_ARGS,                  // Size ≤ 40
+    NUMBER_OF_INDICES,
 };
 
+static constexpr std::string_view AZTEC_DOMAIN = "__AZTEC_";
+static constexpr std::string generatorIndexDomain(GeneratorIndex idx)
+{
```

Comment:
This concept of string domain separation was removed as its not supported in Noir, just yet. I think we only need one Aztec_Domain separator and the rest can be hash indices or we can just use strings -- I believe there was a TODO(suyash) in the cpp code about this.

Once we resolve that TODO, we can add back which ever way we choose and make it supported in Noir